### PR TITLE
News management/create

### PR DESCRIPTION
**Pull request by Veaxx**
# Description

Please include a summary of the change, which issue is fixed and the context if the new feature. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant and add new options if necessary. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: backend/routes/api/newsArticle/create.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+const express = require('express');
+const router = express.Router();
+const NewsArticle = require('../../../models/newsArticle');
+const Image = require('../../../models/image');
+const multer = require('multer');
+const upload = multer({ dest: 'uploads/' });
+const fs = require('fs');
+
+// set up rate limiter: maximum of five requests per minute
+var RateLimit = require('express-rate-limit');
```
Comment (GXZheng312) [inline backend/routes/api/newsArticle/create.js:10]: We hebben al een request limiter middleware.

File: backend/routes/api/newsArticle/create.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+});
+
+router.use(express.json());
+router.use(limiter);
+
+router.get('/', async (req, res) => {
+    res.send('Nieuwsartikel aanmaken');
+});
+
+router.post('/', upload.single('bannerImage'), async (req, res) => {
```
Comment (GXZheng312) [inline backend/routes/api/newsArticle/create.js:23]:
```suggestion
router.post('/', authenticateToken, upload.single('bannerImage'), async (req, res) => {
```

File: management/src/routing.js [hunk 3]
```diff
@@ -11,13 +12,18 @@ const Routing = () => {
   return (
     <DefaultLayout>
       <Routes>
-        <Route path="/" element={<Home />} />
+        <Route element={<ProtectedRoute />}>
+          <Route path="/" element={<Home />} />
```
Comment (GXZheng312) [inline management/src/routing.js:16]: er kunnen meerdere Route in een ProtectedRoute zitten ipv alle route zijn eigen ProtectedRoute bijvoorbeeld:
https://stackoverflow.com/questions/66289122/how-to-create-a-protected-route-with-react-router-dom

**Review by GXZheng312 — CHANGES_REQUESTED** [review]
Ik neem aan dat de moesland-app code door xander is gemaakt?

File: backend/routes/api/newsArticle/update.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+const express = require('express');
+const { body, validationResult } = require('express-validator');
+const router = express.Router();
+const { updateNewsArticleByID } = require('../../../repository/newsArticle');
+
+router.use(express.json());
+
+router.post('/', [
```
[the remark this answers is not in the document]
Comment (Veaxx) [reply backend/routes/api/newsArticle/update.js]: Niet mijn deel

File: management/src/routing.js (hunk 3 above)
Comment (Veaxx) [reply management/src/routing.js:16]: Done

File: management/src/pages/articles/create.js [hunk 5]
```diff
@@ -1,65 +1,103 @@
...
+        reader.readAsDataURL(file);
+    };
 
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        
+        const delta = quillRef.current.getEditor().getContents();
+        const content = JSON.stringify(delta);
+
+        const imageFile = document.querySelector('input[name="image"]').files[0];
```
[the remark this answers is not in the document]
Comment (Veaxx) [reply management/src/pages/articles/create.js:39]: Nog niet direct gelukt. Dus laat ik staan

**Review by GXZheng312 — APPROVED** [review]

**Review by xandermonk — APPROVED** [review]

**Pull request merged**